src/VisualStudio/LiveShare/Impl/Client/Completion/RoslynCompletionService.cs

```diff
@@ -1,33 +0,0 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
-
-using System;
-using System.Collections.Immutable;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.Completion;
-using Microsoft.CodeAnalysis.Options;
-using Microsoft.CodeAnalysis.Text;
-
-namespace Microsoft.VisualStudio.LanguageServices.LiveShare.Client.Completion
-{
-    internal class RoslynCompletionService : CompletionServiceWithProviders
-    {
-        private readonly CompletionService _originalService;
-        private readonly string _language;
-
-        public RoslynCompletionService(Workspace workspace, CompletionService originalService, string language)
-            : base(workspace)
-        {
-            _originalService = originalService ?? throw new ArgumentNullException(nameof(originalService));
-            _language = language ?? throw new ArgumentNullException(nameof(language));
-            workspace.Options = workspace.Options.WithChangedOption(CompletionOptions.BlockForCompletionItems, language, false);
```

Comment:
Are we losing the non-blocking behavior here somewhere? Or are LSP implementations automatically non-blocking?

Reply:
LSP client completion will not block